python/ml4ir/base/io/spark_io.py

```diff
@@ -1,119 +1,227 @@
+
+        Returns:
+            pandas dataframe
+        """
+        self.log("Reading dataframe from : {}".format(infile))
+        return (
+            self.spark_session.read.format("csv")
+            .option("header", "true")
+            .option("inferschema", "true")
+            .option("mode", "DROPMALFORMED")
+            .option("mergeSchema", "true")
+            .load(infile)
+            .toPandas()
         )
 
-    return SparkConfigHolder.HDFS
-
-
-def get_local_fs():
-    """Get local FileSystem handler"""
-    if not SparkConfigHolder.LOCAL_FS:
-        SparkConfigHolder.LOCAL_FS = get_hdfs().getLocal(get_hadoop_config())
-
-    return SparkConfigHolder.LOCAL_FS
-
-
-def read_df(infile):
-    """Read spark dataframe"""
-    return (
-        get_spark_session()
-        .read.format("csv")
-        .option("header", "true")
-        .option("inferschema", "true")
-        .option("mode", "DROPMALFORMED")
-        .option("mergeSchema", "true")
-        .load(infile)
-    )
-
-
-def read_text_file(infile):
-    """Read contents of text file from HDFS using Spark"""
-    return "\n".join(
-        [row["value"] for row in get_spark_session().read.format("text").load(infile).collect()]
-    )
-
-
-def read_json(infile):
-    """Read JSON file as a dictionary from HDFS"""
-    return json.loads(read_text_file(infile))
-
-
-def read_yaml(infile):
-    """Read YAML file as a dictionary from HDFS"""
-    return yaml.safe_load(read_text_file(infile))
-
-
-def copy_from_hdfs(src, dest, logger):
-    """
-    Copy a directory/file from HDFS to local filesystem
-
-    Args:
-        - src: String path to source(on HDFS)
-        - dest: String path to destination(on local file system)
-        - logger: logging handler
-    """
-    if logger:
-        logger.info("Copying files from {} to {}".format(src, dest))
-
-    get_hdfs().copyToLocalFile(get_path_from_str(src), get_path_from_str(dest))
-
-
-def copy_to_hdfs(src, dest, overwrite=True, logger=None):
-    """
-    Copy a directory/file to HDFS from local filesystem
-
-    Args:
-        - src: String path to source(on local file system)
-        - dest: String path to destination(on HDFS)
-        - overwrite: Boolean to specify whether existing destination files should be overwritten
-        - logger: logging handler
-    """
-    if logger:
-        logger.info("Copying files from {} to {}".format(src, dest))
-
-    if overwrite and get_hdfs().exists(get_path_from_str(dest)):
-        get_hdfs().delete(get_path_from_str(dest), True)
+    def read_df_list(self, infiles, sep=",", index_col=None) -> pd.DataFrame:
+        """
+        Load a pandas dataframe from a list of files
+
+        Args:
+            infiles: paths to the csv input files; can be hdfs paths
+            sep: separator to use for loading file
+            index_col: column to be used as index
+
+        Returns:
+            pandas dataframe
+        """
+        raise NotImplementedError
```

Comment:
SparkIO inherits from FileIO, isn't it redundant to overwrite notImplemented methods that were defined there? Any particular reason for it? it occurs a few times in SparkIO.

Reply:
Another question (if you know): just passing the list of files, wouldn't do the trick? Like: 
```
self.spark_session.read.csv
            .option("header", "true")
            .option("inferschema", "true")
            .option("mode", "DROPMALFORMED")
            .option("mergeSchema", "true")
            .load(infiles)
            .toPandas()
```
I am looking [at this question](https://stackoverflow.com/questions/37639956/how-to-import-multiple-csv-files-in-a-single-load).

Reply:
Added.